include/git2/types.h

```diff
@@ -68,17 +68,17 @@ typedef int64_t git_time_t;
 
 /** Basic type (loose or packed) of any Git object. */
 typedef enum {
-	GIT_OBJ_ANY = -2,		/**< Object can be any of the following */
-	GIT_OBJ_BAD = -1,		/**< Object is invalid. */
-	GIT_OBJ__EXT1 = 0,		/**< Reserved for future use. */
-	GIT_OBJ_COMMIT = 1,		/**< A commit object. */
-	GIT_OBJ_TREE = 2,		/**< A tree (directory listing) object. */
-	GIT_OBJ_BLOB = 3,		/**< A file revision object. */
-	GIT_OBJ_TAG = 4,		/**< An annotated tag object. */
-	GIT_OBJ__EXT2 = 5,		/**< Reserved for future use. */
-	GIT_OBJ_OFS_DELTA = 6, /**< A delta, base is given by an offset. */
-	GIT_OBJ_REF_DELTA = 7, /**< A delta, base is given by object id. */
-} git_otype;
+	GIT_OBJECT_ANY =      -2, /**< Object can be any of the following */
+	GIT_OBJECT_BAD =      -1, /**< Object is invalid. */
+	GIT_OBJECT__EXT1 =     0, /**< Reserved. */
+	GIT_OBJECT_COMMIT =    1, /**< A commit object. */
+	GIT_OBJECT_TREE =      2, /**< A tree (directory listing) object. */
+	GIT_OBJECT_BLOB =      3, /**< A file revision object. */
+	GIT_OBJECT_TAG =       4, /**< An annotated tag object. */
+	GIT_OBJECT__EXT2 =     5, /**< Reserved. */
+	GIT_OBJECT_OFS_DELTA = 6, /**< A delta, base is given by an offset. */
+	GIT_OBJECT_REF_DELTA = 7, /**< A delta, base is given by object id. */
+} git_object_t;
```

Comment:
I'm not sure I like `git_object_t`. For me, `_t` usually just says "This is a typedef" instead of "This is a type". So I'd rather vote for either `git_object_type` or `git_object_type_t`.

Reply:
Yeah, I struggled with this one a bit.  I agree with you but we're reasonably consistent with `_t` meaning "this is a type`".  A few examples:

* [`git_ref_t`](https://github.com/libgit2/libgit2/blob/master/include/git2/types.h#L202)
* [`git_branch_t`](https://github.com/libgit2/libgit2/blob/master/include/git2/types.h#L209)
* [`git_filemode_t`](https://github.com/libgit2/libgit2/blob/master/include/git2/types.h#L219)
* [`git_cert_t`](https://github.com/libgit2/libgit2/blob/master/include/git2/types.h#L316)
* [`git_submodule_update_t `](https://github.com/libgit2/libgit2/blob/master/include/git2/types.h#L377)
...

(It's not perfect, eg `git_credtype_t`, which does have values of `GIT_CREDTYPE_...` so that's nice, but they might have been more consistent as `GIT_CREDENTIAL_USERNAME`...)  But anyway, this feels consistent even if it wasn't my preference to begin with.

Reply:
Fair enough :+1: